python/mxnet/context.py

```diff
@@ -258,6 +258,30 @@ def num_gpus():
     check_call(_LIB.MXGetGPUCount(ctypes.byref(count)))
     return count.value
 
+def gpu_memory_info(device_id=0):
+    """Query CUDA for the free and total bytes of GPU global memory.
+
+    Parameters
+    ----------
+    device_id : int, optional
+        The device id of the GPU device.
+
+    Raises
+    ------
+    Will raise an exception on any CUDA error.
+
+    Returns
+    -------
+    (free, total) : (int, int)
```

Comment:
Minor: 'total' - is it referring to total used, total available or the total size of the physical GPU

Reply:
I prefer to leave this as is.  Regarding int vs long, I'm not a Python wizard, but ints are 'plain integers' and longs have unlimited precision:
```
$ python
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> x  = sys.maxsize
>>> x
9223372036854775807
>>> type(x)
<type 'int'>
>>> y=2*x
>>> y
18446744073709551614L
>>> type(y)
<type 'long'>
>>> 
```
And unfortunately, there's not a real short answer to what 'total' memory means.  We're wrapping the cuda call cudaMemGetInfo(), and the NVIDIA documentation says:
```
Returns in *free and *total respectively, the free and total amount of memory available for allocation by the device in bytes.
```
Let's say you've got a GPU with published memory T.  The GPU driver puts some control structures like the page table in that memory, so call that driver overhead D.  Finally, your GPU may be driving a monitor, so a window manager is using the GPU with overhead W.  So what does the API return for 'total' in this scenario?  The answer is T - D.  The long answer then is: 'total' means the total memory available to both your MXNet process **and** other processes that may be using the GPU.  I don't know a way to suggest this succinctly without introducing more confusion.